lib/extract-templates.js

```diff
@@ -0,0 +1,133 @@
+import { parseTemplates } from 'ember-template-imports/lib/parse-templates.js';
+
+export const SUPPORTED_EXTENSIONS = ['.js', '.ts', '.gjs', '.gts'];
```

Comment:
Does `ember-template-imports/lib/parse-template.js` actually parse `hbs` and `precompileTemplate` from `.js`/`.ts` files? If not, I don't think we should add `.js`/`.ts` here just yet.

Reply:
@NullVoxPopuli and @chriskrycho are probably going to know better than me, but as far as I understand [`parseTemplates`](https://github.com/ember-template-imports/ember-template-imports/blob/master/src/parse-templates.ts) is currently able to parse content inside `<template></template>` and ``hbs` ` ``.

Reply:
Yup. Is Handy for folks manually using setComponentTemplate (pre `<template>`)

Reply:
Hmm. Doesn't actually seem to have any direct tests for `parseTemplates` which makes me very uncomfortable with the assertion that it works for all cases of content you might have as templates in `.js`/`.ts` files.

Reply:
I created a super simple test harness over in https://github.com/ember-template-imports/ember-template-imports/pull/36 so we can continue to flesh things out, but as you can see from those tests if we use `parseTemplates` directly like this we have a **TON** of problems. It will identify _**every**_ tagged template expression found in the file!!